components/tools/OmeroPy/src/omero/plugins/ldap.py

```diff
@@ -163,7 +166,15 @@ def getdn(self, args):
         if dn is not None and dn.strip():
             self.ctx.out(dn)
 
-    @admin_only
+    @admin_only(AdminPrivilegeModifyGroup)
+    def update_group(self, iupdate, group):
+        iupdate.saveObject(group)
+
+    @admin_only(AdminPrivilegeModifyUser)
+    def update_user(self, iupdate, user):
+        iupdate.saveObject(user)
+
+    @admin_only()
```

Comment:
Few things:

 * I don't see a particular concern about adding API for `check_admin` as long as it's documented, re-usable, etc
 * Not sure that we're necessarily _keen_ to use the decorator. Probably the biggest downside of _not_ having it, is that it's easier to miss the what's going on. But unless we want to add OR-semantics to `admin_only`, this particular method seems to not match what the decorator provides.
 * All that being said: we may not need to change _anything_ since the internal use of `update_group` and `update_user` will _also_ be wrapped by the decorator usages on those methods.